tests/components/smarthab/test_config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Test the SmartHab config flow."""
+import pysmarthab
+
+from homeassistant import config_entries, setup
+from homeassistant.components.smarthab import DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+
+from tests.async_mock import patch
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("pysmarthab.SmartHab.async_login"), patch(
+        "pysmarthab.SmartHab.is_logged_in", return_value=True
+    ), patch(
+        "homeassistant.components.smarthab.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.smarthab.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_EMAIL: "mock@example.com", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "mock@example.com"
+    assert result2["data"] == {
+        CONF_EMAIL: "mock@example.com",
+        CONF_PASSWORD: "test-password",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)